rosbag2_tests/test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp

```diff
@@ -574,7 +574,7 @@ TEST_F(RecordFixture, record_fails_if_neither_all_nor_topic_list_are_specified)
   auto output = internal::GetCapturedStderr();
 
   EXPECT_THAT(exit_code, Eq(EXIT_FAILURE));
-  EXPECT_THAT(output, HasSubstr("Invalid choice: Must specify topic(s) or -a"));
+  EXPECT_THAT(output, HasSubstr("Invalid choice: Must specify topic(s), regex or -a"));
```

Comment:
see prior comment on "change detector"

Reply:
I have changed both to only check for whether the string is empty. There are a few other tests that also check the exact string, but I believe it would be incorrect to include them in this PR since it otherwise does not change related functionalities.

Reply:
I agree that they shouldn't be changed as part of this review, but it's best to avoid copying bad patterns